Server/src/controllers/authenticationController.ts

```diff
@@ -0,0 +1,96 @@
+import { Request, Response, NextFunction } from 'express';
+import { AuthenticationService } from '../services/authenticationService';
+
+import { IResponse } from '../genericInterfaces/ResponsesInterface'
+import { ISignUpInformation } from '../genericInterfaces/AuthenticationInterfaces';
+import { ILoginInformation } from '../genericInterfaces/AuthenticationInterfaces';
+
+/**
+ * This controller is responsible for verifying the user request has correct parameters input.
+ * After request is verified, the appropriate service can be called to fulfill user signup or login
+ */
+export class AuthenticationController {
+    private authenticationService: AuthenticationService;
+    private invalidResponse: boolean;
+
+    constructor() {
+    }
+
+    createSignUpRequest(request: Request, response: Response, next: NextFunction): Response {
+        this.invalidResponse = this.validateSignUpRequest(request);
+        if (this.invalidResponse) {
+            return response.status(400).json("Request is invalid. Missing attributes")
+        }
+        else {
+            let signUpInfo: ISignUpInformation;
+            signUpInfo = {} as any;
+            signUpInfo.email = request.query.email as string;
```

Comment:
im pretty sure you could spread this object too like so: signUpInfo = { ...request.query }